StormyHares/BangazonApi
# Customer payment pay type access mw

## Ticket Link:
  - #1 #31 

## Description:
  - Developers should be able to get a list of employees and a single employee

## Steps to Test:

### Testing Customer #1
#### SETUP
1.  git fetch 
  1. git checkout CustomerPaymentPayTypeAccess-mw
1. In terminal: `dotnet run`
  1. open Postman

#### TESTING POST
  1. Use the POST method in the drop down method with the url `http://localhost:5000/api/customer` in the text box.
1. In the body, make sure the "raw" and "JSON (application/json) settings are selected and enter the JSON object shown below for the customer.
`{
        "firstName": "Rang",
        "lastName": "Dipkin",
        "lastActive": "12/15/2010",
        "product": null
    }`
1. Repeat steps 2 and 3 with a new JSON object.
`{
        "firstName": "Bilb",
        "lastName": "Ono",
        "lastActive": "12/15/1969",
        "product": null
    }`
- [x] Confirm that there are no errors and the object you posted displays in the lower text box with a unique integer as the `id`. 

#### TESTING GET
1. Change the CRUD method in the drop-down list to GET and press send.
- [x] Confirm that the lower text box contains the 2 JSON objects you posted in the test above. 
- [x] Confirm that the `lastActive` data is formatted in the proper dateTime format which matches the actual date you entered. 

2. Change the url to `http://localhost:5000/api/customer/1`
- [x] Confirm that the object for Rang Dipkin is returned in the lower text box.

3.  Change the url to `http://localhost:5000/api/customer/2`
- [x] Confirm that the object for Bilb Ono is returned in the lower text box.

 #### TESTING PUT
1.  Change the CRUD method in the drop-down list to PUT
1. Change the url to `http://localhost:5000/api/customer/1`
1. Enter the JSON object below into the body text box:
`{
"id": 1, "firstName": "Hosta",
        "lastName": "Mahogey",
        "lastActive": "10/10/2010",
        "product": null
    }`
1. Click `SEND`
- [x] Confirm that the status of `204 No Content` is displayed above the lower text box, and the text box is empty. 

5. Change the CRUD method in the drip-down list to GET
6. Press send.
- [x] Confirm that the returned object in the lower text box is the same as the object below:
`{
    "id": 1,
    "firstName": "Hosta",
    "lastName": "Mahogey",
    "lastActive": "2010-10-10T00:00:00",
    "product": null
}

### Testing PaymentType #31

#### TESTING POST
  1. Use the POST method in the drop down method with the url `http://localhost:5000/api/paymenttype` in the text box.
1.  In the body, enter the JSON object shown below for the customer.
`{
        "description": "Visa"
    }`
1. Press `SEND`
1. Repeat steps 2 and 3 with a new JSON object.
`{
        "description": "PayPal",
    }`
- [x] Confirm that there are no errors and the object you posted displays in the lower text box with a unique integer as the `id`.

#### TESTING GET
1. Change the CRUD method in the drop-down list to GET and press send.
- [x] Confirm that the lower text box contains the 2 JSON objects you posted in the test above. 

2. Change the url to `http://localhost:5000/api/paymenttype/1`
- [x] Confirm that the object for Visa is returned in the lower text box.

3.  Change the url to `http://localhost:5000/api/paymenttype/2`
- [x] Confirm that the object for PayPal is returned in the lower text box.

 #### TESTING PUT
1.  Change the CRUD method in the drop-down list to PUT
1. Change the url to `http://localhost:5000/api/paymenttype/1`
1. Enter the JSON object below into the body text box:
`{
"id": 1, "description": "BitCoin Wallet"
    }`
1. Click `SEND`
- [x] Confirm that the status of `204 No Content` is displayed above the lower text box, and the text box is empty. 

5. Change the CRUD method in the drip-down list to GET
6. Press send.
- [x] Confirm that the returned object in the lower text box is the same as the object below:
{
    "id": 1,
    "description": "BitCoin Wallet"
}

### Testing Payment #31

#### TESTING POST
  1. Use the POST method in the drop down method with the url `http://localhost:5000/api/payment` in the text box.
1.  In the body, enter the JSON object shown below for the customer.
`{
	"accountNumber" : 1111,
	"customerId" : 1,
	"paymentTypeId" : 1
    }`
1. Press `SEND`
1. Repeat steps 2 and 3 with a new JSON object.
`{
	"accountNumber" : 2222,
	"customerId" : 2,
	"paymentTypeId" : 2
    }`
- [x] Confirm that there are no errors and the object you posted displays in the lower text box with a unique integer as the `id`.

#### TESTING GET
1. Change the CRUD method in the drop-down list to GET and press send.
- [x] Confirm that the lower text box contains the 2 JSON objects you posted in the test above. 

2. Change the url to `http://localhost:5000/api/payment/1`
- [x] Confirm that the object for account 1111 is returned in the lower text box.

3.  Change the url to `http://localhost:5000/api/payment/2`
- [x] Confirm that the object for account 2222 is returned in the lower text box.

 #### TESTING PUT
1.  Change the CRUD method in the drop-down list to PUT
1. Change the url to `http://localhost:5000/api/payment/1`
1. Enter the JSON object below into the body text box:
`{
"accountNumber" : 1234,
	"customerId" : 2,
	"paymentTypeId" : 2
}`
1. Click `SEND`
- [x] Confirm that the status of `204 No Content` is displayed above the lower text box, and the text box is empty. 

5. Change the CRUD method in the drip-down list to GET
6. Press send.
- [x] Confirm that the returned object in the lower text box is the same as the object below:
{
    "id": 1,
    "accountNumber" : 1234,
     "customerId" : 2,
     "paymentTypeId" : 2
}


 #### TESTING DELETE

1. Change the CRUD method in the drip-down list to DELETE
1. Change the url to `http://localhost:5000/api/payment/1`
1. Click `SEND`
- [x] Confirm that the status of `200 OK` is displayed above the lower text box, and the text box is empty. 
- [x] Confirm that the object below is in the lower text box:
{
    "id": 1,
    "accountNumber" : 1234,
     "customerId" : 2,
     "paymentTypeId" : 2
}
4. Change the CRUD method in the drip-down list to GET
5. Press send.
- [x] Confirm that only the object with an id = 2 is returned in the lower text box.
